src/qutip_qip/decompose/_utility.py

```diff
@@ -29,3 +30,180 @@ def check_gate(gate, num_qubits):
         raise ValueError("Input is not unitary.")
     if gate.dims != [[2] * num_qubits] * 2:
         raise ValueError(f"Input is not a unitary on {num_qubits} qubits.")
+
+
+def _binary_sequence(num_qubits):
+    """ Defines the binary sequence list for basis vectors of a n-qubit gate.
+
+    The string at index `i` is also the row/column index for a basis vector in
+    a numpy array.
+    """
```

Comment:
There is a compact python function that can generate binary combinations. https://stackoverflow.com/questions/14931769/how-to-get-all-combination-of-n-binary-value
But I think yours is more memory friendly as there is no nested list. Actually, a recursive function looks neater: https://stackoverflow.com/questions/64890117/what-is-the-best-way-to-generate-all-binary-strings-of-the-given-length-in-pytho. 

Just FYI, no need to change unless there is a big difference in efficiency.

Reply:
However, if this function is going to be called many times with the same value, it might be worth to cache the result.

Actually when you compute it for `num_qubits=N`, you already calculated all result for `num_qubits<N`. That's why a recursive function here might be more useful to cache the result.

But maybe not of a very high priority.

Reply:
Ok, it seems that for each unitary decomposition, you only need to call this once. So the overhead is ok.